pages/blog/asynchronously-exporting-data-to-remote-systems.mdx

```diff
@@ -0,0 +1,51 @@
+
+## Working with Asynchronous Destinations
+
+[Eloqua's Bulk API](https://docs.oracle.com/en/cloud/saas/marketing/eloqua-develop/Developers/BulkAPI/bulk-API.htm) is an example of a destination that operates in this "asynchronous" fashion. Importing data to Eloqua involves a few steps:
+
+1. **Create an import**. This involves specifying which fields you want to import, and gives you a URI that you can use to upload the data.
+2. **Upload the data.** Using the URI retrieved in the previous step, the data you want to import is uploaded to a "staging area".
+3. **Initiate the import.** This request actually triggers the import to be executed. Instead of directly receiving a response with the results of the import, you're given a URI that can be used to check on the status import later on.
+
+Similarly, [SendGrid](https://sendgrid.api-docs.io/v3.0/contacts/add-or-update-a-contact) can also work like this, though it's a bit more streamlined and only requires a single request to set up the import. The result of this request, much like Eloqua, includes a `job_id` that you can use to check on the status of the import.
+
+### Size limits
+
+Something to note about these imports is that they usually have a limit on the size of the request that can be issued. For example, Eloqua has a hard limit of 32MB per request, though it allows you to issue multiple upload requests up to a recommended 250MB in total. SendGrid is a bit more limiting, only allowing 6MB of data to be sent per job.
+
+### Checking on completion
+
+Some of these APIs allow you to give them a `callbackUrl` that will be called once an import is complete. This may be a good option if you're able to expose an endpoint that the remote system can call, but this isn't always possible or convenient depending on your setup.
+
+The more common approach is to receive an ID that can be used to check on how the import is doing via some status endpoint. This requires you to implement a recurring polling solution to keep checking until it's done.
+
+These endpoints usually return a `status` that indicates where in the process a given import is at. For example, Eloqua's imports can be `pending`, `active`, `success`, `warning` or `errored`.
+
+### Handling errors
+
+An interesting consequence of having this two-step process is that now errors need to be handled at different stages. In the initial upload of the data, some APIs can return validation errors for some profiles. Then, when the import is actually completed, some profiles may have failed, which also needs to be handled. The action of checking on the import could also potentially fail, which also needs to be considered.
+
+## Grouparoo
+
+As a tool to keep your data synchronized between systems, Grouparoo has all of this built-in. Take a look at our available integrations, or make your own plugin to leverage the platform. Best of all, we're open source!
```

Comment:
Take a look at our available [integrations](/integrations)

Reply:
Grouparoo has all of (this) built-in. ->  Grouparoo has support for all types of destinations, including async destinations, this built-in.